src/persist-client/src/internal/compact.rs

```diff
@@ -921,4 +1031,96 @@ mod tests {
         assert_eq!(part.desc, res.output.desc);
         assert_eq!(updates, all_ok(&data, 10));
     }
+
+    #[tokio::test]
+    async fn prefetches() {
+        let desc = Description::new(
+            Antichain::from_elem(0u64),
+            Antichain::new(),
+            Antichain::new(),
+        );
+        let input_parts = (0..=9)
+            .map(|encoded_size_bytes| HollowBatchPart {
+                key: PartialBatchKey("".into()),
+                encoded_size_bytes,
+            })
+            .collect::<Vec<_>>();
+        let parse = |x: &str| {
+            x.split('|')
+                .map(|run| {
+                    let parts = run
+                        .split(',')
+                        .map(|x| {
+                            let encoded_size_bytes = x.get(1..).unwrap().parse::<usize>().unwrap();
+                            match x.get(..1).unwrap() {
+                                " " => CompactionPart::Queued(&input_parts[encoded_size_bytes]),
+                                "f" => CompactionPart::Prefetched(
+                                    encoded_size_bytes,
+                                    spawn(|| "", async {
+                                        tokio::time::sleep(Duration::from_secs(1_000_000)).await;
+                                        unreachable!("boom")
+                                    }),
+                                ),
+                                x => panic!("unknown {}", x),
+                            }
+                        })
+                        .collect::<VecDeque<_>>();
+                    (&desc, parts)
+                })
+                .collect::<Vec<_>>()
+        };
+        let print = |x: &Vec<(&Description<u64>, VecDeque<CompactionPart<'_, u64>>)>| {
+            x.iter()
+                .map(|(_, x)| {
+                    x.iter()
+                        .map(|x| match x {
+                            CompactionPart::Queued(x) => format!(" {}", x.encoded_size_bytes),
+                            CompactionPart::Prefetched(x, _) => format!("f{}", x),
+                        })
+                        .collect::<Vec<_>>()
+                        .join(",")
+                })
+                .collect::<Vec<_>>()
+                .join("|")
+        };
+
+        let client = new_test_client().await;
+
+        let shard_id = ShardId::new();
+        let blob = &client.blob;
+        let metrics = &client.metrics;
+
+        // Enough budget for none, some, and all parts
```

Comment:
```suggestion
        // Enough budget for none, some, and all parts of a single run
```

Reply:
(at a glance, I had trouble telling whether `,` or `|` indicated part vs run. might be helpful if the initial comments here added a bit more context)